contracts/apps/AppStorage.sol

```diff
@@ -4,14 +4,14 @@
 
 pragma solidity ^0.4.18;
 
-import "../kernel/IKernel.sol";
+import "../common/UnstructuredStorage.sol";
 
 
-contract AppStorage {
-    IKernel public kernel;
-    bytes32 public appId;
-    address internal pinnedCode; // used by Proxy Pinned
-    uint256 internal initializationBlock; // used by Initializable
-    uint256[95] private storageOffset; // forces App storage to start at after 100 slots
-    uint256 private offset;
+contract AppStorage is UnstructuredStorage {
+    // keccak256("IKernel.kernel")
```

Comment:
I would remove the type from the key names as types can change in the future and they are all storing a 32-byte word anyway.

It would make sense to have the type be explicit if we were storing more complex data types as arrays, mappings or structs, but for simple values I would remove it.

Reply:
I would also add an `aragonOS.appStorage` prefix to all keys to remove the possibility of collisions.